dimod/generators/constraints.py

```diff
@@ -91,9 +92,8 @@ def combinations(n, k, strength=1, vartype=BINARY):
     lbias = float(strength*(1 - 2*k))
     qbias = float(2*strength)
 
-    bqm = BinaryQuadraticModel.empty(BINARY)
-    bqm.add_variables_from(((v, lbias) for v in variables))
-    bqm.add_interactions_from(((u, v, qbias) for u, v in itertools.combinations(variables, 2)))
-    bqm.offset += strength*(k**2)
+    Q = np.triu(np.ones((n,n))*qbias, k=1)
+    np.fill_diagonal(Q, lbias)
+    bqm = BinaryQuadraticModel(Q, 'BINARY', offset=strength*(k**2))
```

Comment:
This doesn't work currently (see [here](https://github.com/dwavesystems/dimod/blob/c69abcf882b2cc78fc124d2733e2a28d02a66e46/dimod/binary/binary_quadratic_model.py#L77-L105)).

Until kwargs/offset is nicely supported, one option is to write it like this:
```
bqm = BinaryQuadraticModel({}, Q, offset, 'BINARY')
```

but much nicer is to use the qubo factory:
```suggestion
    bqm = BinaryQuadraticModel.from_qubo(Q, offset=strength*(k**2))
```